sql/api/src/main/scala/org/apache/spark/sql/catalyst/trees/origin.scala

```diff
@@ -76,6 +76,12 @@ object CurrentOrigin {
       value.get.copy(line = Some(line), startPosition = Some(start)))
   }
 
+  def setPySparkErrorContext(pysparkFragment: String, pysparkCallSite: String): Unit = {
+    val tupleInfo = (pysparkFragment, pysparkCallSite)
+    value.set(
+      value.get.copy(pysparkErrorContext = Some(tupleInfo)))
```

Comment:
now JVM stacktrace and python stacktrace can't coexist?

Reply:
I belive we still keeping JVM stacktrace since we're using `copy` here. That means the other parameters are still there, but just updating the `pysparkErrorContext` for the existing `CurrentOrigin`.